packages/nimble-components/src/wafer-map/modules/wafer-map-update-tracker.ts

```diff
@@ -93,16 +93,26 @@ export class WaferMapUpdateTracker extends UpdateTracker<typeof trackedItems> {
      * After the update is finished, all the tracked items are reset.
      */
     public override queueUpdate(): void {
-        if (!this.wafermap.$fastController.isConnected) {
+        if (!this.wafermap.$fastController.isConnected || this.updateQueued) {
             return;
         }
-        if (!this.updateQueued) {
-            this.updateQueued = true;
+        this.updateQueued = true;
+        if (this.wafermap.currentTask === undefined) {
             DOM.queueUpdate(() => {
                 this.wafermap.update();
                 this.untrackAll();
                 this.updateQueued = false;
             });
+        } else {
+            void (async () => {
+                await this.wafermap.currentTask;
```

Comment:
The [timeline](https://github.com/ni/nimble/issues/2034) for `Render Dies` states "error handling of rejection" however there is no error handling of rejection in this PR (or tests for failure states).

i.e. the `await this.wafermap.currentTask` line can `throw` do to an Error being thrown in the worker, etc. But it's not being handled here.

@Razvan1928 @munteannatan The pattern we discussed is at least initially reporting the error to users via the validity flags. The options I can think of are:
- address it in this PR and then check the `Render Dies` task
- do not address it in this PR (and do not close the `Render Dies` task) but make a follow-up PR that does do the error handling to finish the `Render Dies` task.

Reply:
As was discussed on teams we will do the 'error handling of rejection' in a different pr